spec/truffle/polyglot/exception_spec.rb

```diff
@@ -0,0 +1,18 @@
+# Copyright (c) 2022 Oracle and/or its affiliates. All rights reserved. This
+# code is released under a tri EPL/GPL/LGPL license. You can use it,
+# redistribute it and/or modify it under the terms of the:
+#
+# Eclipse Public License version 2.0, or
+# GNU General Public License version 2, or
+# GNU Lesser General Public License version 2.1.
+# OTHER DEALINGS IN THE SOFTWARE.
+
+require_relative '../../ruby/spec_helper'
+
+describe 'Polygot::ForeignException' do
+  it 'cannot be marshaled' do
+    -> {
+      Marshal.dump(Polyglot::ForeignException.new 'External LLVMFunction rb_enc_strlen cannot be found.')
```

Comment:
We should probably forbid calling new/allocate on Polyglot::ForeignObject because that's actually not quite the same thing as a truly foreign object/exception (e.g. which are not represented with RubyDynamicObject).
Could you use `Truffle::Debug.foreign_exception` instead?